src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -307,20 +315,26 @@ public IndexFieldsMapping GetKnownFields()
         }
     }
     
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     void HandleObject(BlittableJsonReaderObject val, IndexField field, JsonOperationContext indexContext, ref IndexEntryWriter entryWriter,
         IWriterScope scope, bool nestedArray = false)
     {
         if (val.TryGetMember(RavenConstants.Json.Fields.Values, out var values) &&
             IsArrayOfTypeValueObject(val))
         {
-            HandleArray((IEnumerable)values, field, indexContext, ref entryWriter, scope, true);
+            InsertRegularField(field, (IEnumerable)values, indexContext, ref entryWriter, scope, nestedArray);
         }
```

Comment:
@ayende @ppekrol  any ideas on how to hit that part of code?
` public const string Values = "$values";`
It comes from Lucene's IndexWriter

Reply:
This is done when you are working on a document that saved an array in this manner:

```
{
   '$type': 'HashSet<string>',
   '$values': [ "1", "2", "Etc"],
}
```